src/Sonarr.Api.V3/Blocklist/BlocklistController.cs

```diff
@@ -23,12 +25,17 @@ public class BlocklistController : Controller
 
         [HttpGet]
         [Produces("application/json")]
-        public PagingResource<BlocklistResource> GetBlocklist([FromQuery] PagingRequestResource paging)
+        public PagingResource<BlocklistResource> GetBlocklist([FromQuery] PagingRequestResource paging, [FromQuery] int[] seriesIds = null, [FromQuery] DownloadProtocol[] protocols = null)
         {
             var pagingResource = new PagingResource<BlocklistResource>(paging);
             var pagingSpec = pagingResource.MapToPagingSpec<BlocklistResource, NzbDrone.Core.Blocklisting.Blocklist>("date", SortDirection.Descending);
 
-            return pagingSpec.ApplyToPage(_blocklistService.Paged, model => BlocklistResourceMapper.MapToResource(model, _formatCalculator));
+            if (seriesIds != null && seriesIds.Any())
+            {
+                pagingSpec.FilterExpressions.Add(b => seriesIds.Contains(b.SeriesId));
+            }
+
```

Comment:
Looks like for protocols we can do the same as series ID filtering:

```

if (protocols != null && protocols.Any())
{
    pagingSpec.FilterExpressions.Add(b => protocols.Contains(b.Protocol));
}
```

Which saves us passing protocol through, unless there is a reason why we wouldn't want to do that?

Reply:
makes sense, I didn't consider this but have moved this to the pagingspec filter instead and removed ther work from the repo